zarr/indexing.py

```diff
@@ -0,0 +1,676 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, print_function, division
+import numbers
+import itertools
+import collections
+
+
+import numpy as np
+
+
+def is_integer(x):
+    return isinstance(x, numbers.Integral)
```

Comment:
Make sure this catches numpy's signed and unsigned integer types -- missing those lead to issues in dask and xarray.

Reply:
I've checked this, looks OK:

```
In [5]: for t in int, np.int8, np.int16, np.int32, np.int64, np.uint8, np.uint16, np.uint32, np.uint64:
   ...:     print(t, isinstance(t(42), numbers.Integral))
   ...:     
<class 'int'> True
<class 'numpy.int8'> True
<class 'numpy.int16'> True
<class 'numpy.int32'> True
<class 'numpy.int64'> True
<class 'numpy.uint8'> True
<class 'numpy.uint16'> True
<class 'numpy.uint32'> True
<class 'numpy.uint64'> True
```